balance/src/lib.rs

```diff
@@ -0,0 +1,351 @@
+extern crate futures;
+extern crate h2;
+extern crate http;
+extern crate tower_balance;
+extern crate tower_h2;
+
+use futures::{Async, Poll};
+use tower_balance::load::Instrument;
+
+/// Instruments HTTP responses to drop handles when their first body message is received.
+#[derive(Clone, Debug, Default)]
+pub struct PendingUntilFirstData(());
+
+/// Instruments HTTP responses to drop handles when their streams completes.
+#[derive(Clone, Debug, Default)]
+pub struct PendingUntilEos(());
+
+/// An instrumented HTTP body that drops its handle when the first data is received.
+pub struct PendingUntilFirstDataBody<T, B> {
+    handle: Option<T>,
+    body: B,
+}
+
+/// An instrumented HTTP body that drops its handle upon completion.
+pub struct PendingUntilEosBody<T, B> {
+    handle: Option<T>,
+    body: B,
+}
+
+// ==== PendingUntilEos ====::default()
+
+impl<T, B> Instrument<T, http::Response<B>> for PendingUntilFirstData
+where
+    T: Sync + Send + 'static,
+    B: tower_h2::Body + 'static,
+{
+    type Output = http::Response<PendingUntilFirstDataBody<T, B>>;
+
+    fn instrument(&self, handle: T, rsp: http::Response<B>) -> Self::Output {
+        let (parts, body) = rsp.into_parts();
+        let handle = if body.is_end_stream() {
+            drop(handle);
```

Comment:
Not sure if this is strictly necessary, as `handle` will drop when the function returns, but maybe the explicit drop is clearer...

Reply:
yes, this was intentional